src/basic/process-util.c

```diff
@@ -1727,6 +1728,43 @@ int make_reaper_process(bool b) {
         return 0;
 }
 
+int posix_spawn_wrapper(const char *path, char *const *argv, char *const *envp, pid_t *ret_pid) {
+        posix_spawnattr_t attr;
+        sigset_t mask;
+        pid_t pid;
+        int r;
+
+        assert(path);
+        assert(argv);
+        assert(ret_pid);
+
+        assert_se(sigemptyset(&mask) >= 0);
+        assert_se(sigaddset(&mask, SIGCHLD) >= 0);
```

Comment:
as mentioned, i still don#t get this. i can#t see where we did this so far, nor what this good for, nor that the effect of this would even survive for long

Reply:
I'm trying to get the new behavior to be the same as the old one. As far as I understand semantics of posix_spawn are different, and if nothing is passed here, it will copy pid1's mask. If that's what we want I'll drop it. If we need something different, what should it be instead? I am asking because I am not sure

Reply:
will drop and just leave empty set as discussed